### ISSUE TITLE
Add rule to check uniqueness of `messageId`

### ISSUE THREAD

**Issue by magicmatatjahu** [stoplightio/spectral#2223]
**User story.**
As a user, I want to know if the messages defined in my AsyncAPI document have a unique `messageId`.

**Describe the solution you'd like**
The following AsyncAPI document is invalid and Spectral should throw errors due to non-uniqueness of `messageId`.

```yaml
asyncapi: '2.4.0'
...

channels:
  someChannel:
    publish:
      message:
        messageId: someId
        ...
    subscribe:
      message:
        messageId: someId
```

**Additional context**
Part of #2100
